### DISCUSSION TITLE
Adapt to run using Open-source / Local structure

### DISCUSSION THREAD

**Discussion by lcjr86** [starpig1129/Multi-agent-DataAnalysis#2] [Ideas]
Hello,
I think that the project is excellent!
Do you have plans to adapt the code to use local/open-source LLMs?

Best regards,

Luiz

**Comment by zi-yue-1129** [comment]
Hello Luiz,

Thank you very much for your  suggestion about the project!

Regarding plans to adapt the code for local/open-source LLMs, I have indeed
considered this direction. However, due to LangChain's limited support for
local models, this may require more development work.

For instance, in another project of mine - a Discord bot
<https://github.com/starpig1129/ai-discord-bot-PigPig> - I've already
implemented local models, but it required significant adjustments to make
the system work properly. Adapting this project to support local models
would likely need similar modifications.

If you're interested in this aspect, I'd welcome your involvement! Perhaps
we could collaborate to develop a version that supports local models. Do
you have any thoughts on this?

I'm looking forward to your reply.

Best regards,

Zi Yue Chao

Luiz Carlos de Jesus Junior ***@***.***> 於 2024年9月10日 週二
下午7:39寫道：

**Reply by sevapru** [reply]
Hi man, I could probably help you local models support. Could you find me anywhere by the same @ that I have in Github :surfer:

**Reply by zi-yue-1129** [reply]
Hi ，Can you send your contact information to my email? 
james911129@gmail.com

**Reply by zi-yue-1129** [reply]
@sevapru

**Comment by lcjr86** [comment]
it will be cool to have that integrated with ollama models, for example ;-)

**Reply by zi-yue-1129** [reply]
I will probably first adjust the non-local version to be better

**Comment by jana37100** [comment]
I also would like to use it using open source models. I spent all day trying to adapt the code to Ollama which can follow the open AI structure but its a little tricky to know exactly where to edit it,

**Reply by zi-yue-1129** [reply]
It may be necessary to stably output the model in json format
And write a class similar to ChatOpenAI

**Reply by jana37100** [reply]
In Ollama you can use similar class to OpenAI with for example:
"from ollama import Client
client = Client(
  host='URL_to_hosted_model',
  headers={'x-some-header': 'some-value'}
)

response = client.chat(model='llama3.1', messages=[
  {
    'role': 'user',
    'content': 'Why is the sky blue?',
  },
])
"

What I was trying to find is where best to subsititute this with what you have in the code, just with main.py, and the main. notebook?

**Reply by zi-yue-1129** [reply]
Maybe you can start modifying it from here
[https://github.com/starpig1129/AI-Data-Analysis-MultiAgent/blob/main/core/language_models.py](https://github.com/starpig1129/AI-Data-Analysis-MultiAgent/blob/main/core/language_models.py)
